core/parachain/pvf/pool.hpp

```diff
@@ -28,11 +31,15 @@ namespace kagome::parachain {
    public:
     PvfPool(const application::AppConfiguration &app_config,
             std::shared_ptr<runtime::ModuleFactory> module_factory,
-            std::shared_ptr<runtime::InstrumentWasm> instrument);
+            std::shared_ptr<runtime::WasmInstrumenter> instrument);
 
-    auto &pool() const {
-      return pool_;
-    }
+    OptRef<const runtime::Module> getModule(
+        const Hash256 &code_hash,
+        const runtime::RuntimeContext::ContextParams &config) const;
+
+    std::filesystem::path getCachePath(
+        const common::Hash256 &code_hash,
+        const runtime::RuntimeContext::ContextParams &config) const;
```

Comment:
Revert to `pool()` getter to avoid duplicating interface

Reply:
It's intentional for encapsulation, to highlight that cached instanced from the underlying pool should not be used for PVFs.